src/simsopt/geo/magneticfieldclasses.py

```
@@ -0,0 +1,116 @@
+       Example: ScalarPotentialRZMagneticField("2*phi") yields a magnetic field B = grad(2*phi) = (0,2/R,0).
+       Note: this function needs sympy.    
+
+    Args:
+        PhiStr:  string containing vacuum scalar potential expression as a function of R, Z and phi
+    '''
+    def __init__(self, PhiStr):
+        if not sympy_found:
+            raise RuntimeError("Sympy is required for the ScalarPotentialRZMagneticField class")
+        self.PhiStr = PhiStr
+        self.Phiparsed = parse_expr(PhiStr)
+        R,Z,Phi = sp.symbols('R Z phi')
+        self.Blambdify = sp.lambdify((R, Z, Phi), [self.Phiparsed.diff(R), self.Phiparsed.diff(Phi)/R, self.Phiparsed.diff(Z)])
+        self.dBlambdify_by_dX = sp.lambdify((R, Z, Phi), 
+                            [[sp.cos(Phi)*self.Phiparsed.diff(R).diff(R)-(sp.sin(Phi)/R)*self.Phiparsed.diff(R).diff(Phi),sp.cos(Phi)*(self.Phiparsed.diff(Phi)/R).diff(R)-(sp.sin(Phi)/R)*(self.Phiparsed.diff(Phi)/R).diff(Phi),sp.cos(Phi)*self.Phiparsed.diff(Z).diff(R)-(sp.sin(Phi)/R)*self.Phiparsed.diff(Z).diff(Phi)],
+                             [sp.sin(Phi)*self.Phiparsed.diff(R).diff(R)+(sp.cos(Phi)/R)*self.Phiparsed.diff(R).diff(Phi),sp.sin(Phi)*(self.Phiparsed.diff(Phi)/R).diff(R)+(sp.cos(Phi)/R)*(self.Phiparsed.diff(Phi)/R).diff(Phi),sp.sin(Phi)*self.Phiparsed.diff(Z).diff(R)+(sp.cos(Phi)/R)*self.Phiparsed.diff(Z).diff(Phi)],
+                             [self.Phiparsed.diff(R).diff(Z),(self.Phiparsed.diff(Phi)/R).diff(Z),self.Phiparsed.diff(Z).diff(Z)]])
+        
+    def compute(self, points, compute_derivatives=0):
+        r   = np.sqrt(np.power(points[:,0],2) + np.power(points[:,1],2))
+        z   = points[:,2]
+        phi = np.arctan2(points[:,1],points[:,0])
+        self._B = [self.Blambdify(r[i],z[i],phi[i]) for i in range(len(r))]
+
+        if compute_derivatives >= 1:
+            self._dB_by_dX = [self.dBlambdify_by_dX(r[i],z[i],phi[i]) for i in range(len(r))]
```

Please raise an error in case a hessian is required